pkg/scm/gitlab/client_actioner.go

```diff
@@ -153,6 +155,81 @@ func (c *Client) ApplyStep(ctx context.Context, evalContext scm.EvalContext, upd
 
 		return err
 
+	case "assign_reviewers":
+		source, err := step.RequiredStringEnum("source", "codeowners")
+		if err != nil {
+			return err
+		}
+
+		desiredLimit, err := step.RequiredInt("limit")
+		if err != nil {
+			return err
+		}
+
+		mode, err := step.OptionalStringEnum("mode", "linear", "linear", "random")
+		if err != nil {
+			return err
+		}
+
+		var eligibleReviewers []scm.Actor
+
+		switch source {
+		case "codeowners":
+			eligibleReviewers = evalContext.GetCodeOwners()
+
+			break
+		}
+
+		if len(eligibleReviewers) == 0 {
+			return nil
+		}
+
+		var reviewers []scm.Actor
+
+		var limit int
+		if desiredLimit > len(eligibleReviewers) {
+			limit = len(eligibleReviewers)
+		} else {
+			limit = desiredLimit
+		}
+
+		switch mode {
+		case "linear":
+			reviewers = eligibleReviewers[:limit]
+
+			break
+		case "random":
+			reviewers = make([]scm.Actor, limit)
+			r := rand.New(rand.NewSource(time.Now().UnixNano()))
+			perm := r.Perm(len(eligibleReviewers))
+
+			for i := 0; i < limit; i++ {
+				reviewers[i] = eligibleReviewers[perm[i]]
+			}
+
+			break
+		}
+
+		var reviewerIDs []int
+
+		for _, reviewer := range reviewers {
+			reviewerIDs = append(reviewerIDs, reviewer.IntID())
```

Comment:
annoying that the gitlab api accepts only int ids instead of the username (or even the string id string it returns for the user) for the update endpoint

Reply:
GitLab has many annoyances and inconsistencies in both REST and GraphQL API :(